modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCachePartitionExchangeManager.java

```diff
@@ -1564,9 +1564,17 @@ else if (!grp.isLocal())
                     log.debug("Notifying exchange future about single message: " + exchFut);
 
                 if (msg.client() && !exchFut.isDone()) {
-                    if (exchFut.initialVersion().compareTo(readyAffinityVersion()) <= 0) {
+                    AffinityTopologyVersion initVer = exchFut.initialVersion();
+                    AffinityTopologyVersion readyVer = readyAffinityVersion();
+
+                    DiscoveryEvent firstEvt = exchFut.firstEvent();
+
+                    if (initVer.compareTo(readyVer) <= 0
+                        && (firstEvt != null && firstEvt.type() != EVT_NODE_LEFT && firstEvt.eventNode().isClient())) {
                         U.warn(log, "Client node tries to connect but its exchange " +
-                            "info is cleaned up from exchange history." +
+                            "info is cleaned up from exchange history " +
```

Comment:
We have a specific string output format described here -- https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines

From my point, we should also use it here. Square, round brackets not at the right places here and should be placed at the end of warn message.

Reply:
fixed